cmd/ibmcloud-janitor-boskos/main.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/logrusutil"
+	boskosClient "sigs.k8s.io/boskos/client"
+	"sigs.k8s.io/boskos/common"
+	"sigs.k8s.io/boskos/ibmcloud-janitor/resources"
+)
+
+var (
+	boskosURL    = flag.String("boskos-url", "", "Boskos URL")
+	rTypes       common.CommaSeparatedStrings
+	username     = flag.String("username", "", "Username used to access the Boskos server")
+	passwordFile = flag.String("password-file", "", "The path to password file used to access the Boskos server")
+	logLevel     = flag.String("log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	debug        = flag.Bool("debug", false, "Setting it to true allows logs for PowerVS client")
+)
+
+const (
+	sleepTime = time.Minute
+)
+
+func init() {
+	flag.Var(&rTypes, "resource-type", "comma-separated list of resources need to be cleaned up")
+}
+
+//nolint:nestif
+func run(boskos *boskosClient.Client) error {
+	for {
+		for _, resourceType := range rTypes {
+			if res, err := boskos.Acquire(resourceType, common.Dirty, common.Cleaning); errors.Cause(err) == boskosClient.ErrNotFound {
```

Comment:
is this `boskos.Acquire` randomly pick or goes in order?

if goes in order what happens in the next loop if it fails to cleanup the resource

Reply:
Acquire follows FIFO approach. The unsuccessfully cleaned up dirty resource will be eventually picked up by janitor in the next loop for cleanup.

Reply:
but happens for the next loop, is that resource again sit first in the queue or added in the end?

Reply:
It might get added to the end of the queue since the resource would transition from `dirty --> cleaning --> dirty`.